ui/components/multichain/asset-picker-amount/asset-picker-modal/asset-picker-modal.tsx

```diff
@@ -0,0 +1,245 @@
+      isSelected: boolean;
+    }) => {
+      token.isSelected =
+        token.address?.toLowerCase() === selectedToken?.toLowerCase();
+      return token;
+    },
+  );
+
+  tokenList.push({
+    address: null,
+    symbol: nativeCurrency,
+    decimals: 18,
+    image: nativeCurrencyImage,
+    balance: balanceValue,
+    string: primaryCurrencyProperties.value,
+    type: AssetType.native,
+    isSelected: selectedToken === null,
+  });
+
+  const tokensData = tokenList.filter((item) =>
+    item.symbol.toLowerCase().includes(searchQuery.toLowerCase()),
+  );
+
+  const handleSearch = (query: string) => {
+    setSearchQuery(query);
+  };
+
+  const handleSelectToken = async (token: any) => {
+    setSelectedToken(token.address);
+    if (token.type === AssetType.native) {
+      await dispatch(
+        updateSendAsset({
+          type: token.type ?? AssetType.native,
+          details: token,
+          skipComputeEstimatedGasLimit: true,
+        }),
+      );
+      history.push(SEND_ROUTE);
+      onClose();
+    }
+    await dispatch(
+      updateSendAsset({
+        type: token.type ?? AssetType.token,
+        details: { ...token, standard: TokenStandard.ERC20 },
+        skipComputeEstimatedGasLimit: true,
+      }),
+    );
+    history.push(SEND_ROUTE);
+    onClose();
+  };
+
+  return (
+    <Modal
+      className="asset-picker-modal"
+      isOpen={isOpen}
+      onClose={onClose}
+      data-testid="asset-picker-modal"
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader onClose={onClose}>{t('selectAToken')}</ModalHeader>
+        <div>
+          <TextFieldSearch
+            placeholder={t('searchTokenOrNFT')}
+            value={searchQuery}
+            onChange={(e: any) => handleSearch(e.target.value)}
+            error={false}
+            autoFocus
+            autoComplete={false}
+            width={BlockSize.Full}
+            clearButtonOnClick={() => setSearchQuery('')}
+            clearButtonProps={{
+              size: Size.SM,
+            }}
+            showClearButton={true}
+            className={'asset-picker-modal__search-list'}
+            inputProps={{
+              'data-testid': 'asset-picker-modal-search-input',
+            }}
+            endAccessory={null}
+          />
+          {tokensData.map((token) => {
+            return (
+              <Box
+                padding={4}
+                gap={2}
+                key={token.symbol}
+                backgroundColor={
+                  token.isSelected
+                    ? BackgroundColor.primaryMuted
+                    : BackgroundColor.transparent
+                }
+                className={classnames('multichain-asset-picker-list-item', {
+                  'multichain-asset-picker-list-item--selected':
+                    token.isSelected,
+                })}
+                onClick={() => handleSelectToken(token)}
+              >
+                {token.isSelected && (
```

Comment:
We've been trying to avoid using `&&` for conditional render.  We try to use:

`condition ? < /> : null`